fix possible divide by zero bug

xiaoxiaoafeifei opened the pull request:
In the file ssl/record/methods/ssl3_meth.c, the function ssl3_mac has the following code：

```
t = EVP_MD_CTX_get_size(hash);
if (t < 0)
    return 0;
md_size = t;
npad = (48 / md_size) * md_size;
```

The variable md_size is used as a divisor. Since the code explicitly checks whether u is negative, I think it should also avoid the case when md_size=0 to prevent divide by zero bugs.

t8m commented:
This would be acceptable with CLA: trivial. Can you please `git commit --amend` the commit message so it contains `CLA: trivial` on a separate line in the commit message body?

t8m commented:
The patch will probably not cherry-pick cleanly to other branches but the problem is there too.

xiaoxiaoafeifei commented:
> This would be acceptable with CLA: trivial. Can you please `git commit --amend` the commit message so it contains `CLA: trivial` on a separate line in the commit message body?

I updated commit message as requested

t8m commented:
> > This would be acceptable with CLA: trivial. Can you please `git commit --amend` the commit message so it contains `CLA: trivial` on a separate line in the commit message body?
> 
> I updated commit message as requested

There must be a space after `:` otherwise the cla-check won't recognize it.

xiaoxiaoafeifei commented:
> > > This would be acceptable with CLA: trivial. Can you please `git commit --amend` the commit message so it contains `CLA: trivial` on a separate line in the commit message body?
> > 
> > 
> > I updated commit message as requested
> 
> There must be a space after `CLA:` otherwise the cla-check won't recognize it.

I have modified it

xiaoxiaoafeifei commented:
> Hmm, something is still wrong - There also needs to be an empty line between the first commit message line (this is the commit message title) and the rest of the commit message (this is the commit message body). Unfortunately GitHub does not present the commit messages in raw form.

I have modified it as follows

`    fix possible divide by zero bug
    
    CLA: trivial
`

Is it ok？

Thank you.

openssl-machine commented:
This pull request is ready to merge

t8m commented:
Merged to master branch. Would you like to create equivalent PR agains the openssl-3.1 branch? Or should I do it?

xiaoxiaoafeifei commented:
> The patch will probably not cherry-pick cleanly to other branches but the problem is there too.


> Merged to master branch. Would you like to create equivalent PR agains the openssl-3.1 branch? Or should I do it?

Sorry, I just saw this message，you can do it if you like.
Thank you
